src/probnum/filtsmooth/particlefiltsmooth/_particle_filter.py

```diff
@@ -0,0 +1,218 @@
+"""Particle filters."""
+
+from typing import Optional, Union
+
+import numpy as np
+
+from probnum import random_variables, statespace
+from probnum.filtsmooth.bayesfiltsmooth import BayesFiltSmooth
+from probnum.type import FloatArgType, IntArgType
+
+from ._particle_filter_posterior import ParticleFilterPosterior
+
+
+def effective_number_of_events(categ_rv: random_variables.Categorical) -> float:
+    """Effective number of events in the support of a categorical random variable.
+
+    In a particle filter, this is used as the effective number of
+    particles.
+    """
+    return 1.0 / np.sum(categ_rv.probabilities ** 2)
```

Comment:
this is not a property of the categorical RV because I dont think it has any meaning outside of using categorical RVs for importance sampling